service/controller/releasecycle/resource/app/desired.go

```diff
@@ -4,84 +4,56 @@ import (
 	"context"
 
 	applicationv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
-	releasev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
 	"github.com/giantswarm/microerror"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/giantswarm/release-operator/pkg/project"
 	"github.com/giantswarm/release-operator/service/controller/key"
 )
 
+// GetDesiredState computes the desired App CR for the release referenced in ReleaseCycle CR.
 func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
 	releaseCycleCR, err := key.ToReleaseCycleCR(obj)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	releaseCR, err := r.g8sClient.ReleaseV1alpha1().Releases(releaseCycleCR.GetNamespace()).Get(releaseCycleCR.Spec.Release.Name, metav1.GetOptions{})
+	appName := releaseAppCRName(releaseCycleCR)
+	r.logger.LogCtx(ctx, "level", "debug", "message", "computing desired state", "app", appName)
+
+	releaseProvider, releaseVersion, err := key.SplitReleaseName(releaseCycleCR.GetName())
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	var appCRs []*applicationv1alpha1.App
-	{
-		r.logger.LogCtx(ctx, "level", "debug", "message", "computing desired state")
-
-		for _, component := range releaseCR.Spec.Components {
-			appCR, err := r.newAppCR(ctx, *releaseCR, component)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
+	releaseRepo := releasePrefix(releaseProvider)
+	appCR := r.newAppCR(appName, releaseRepo, releaseVersion)
 
-			appCRs = append(appCRs, appCR)
-		}
-
-		r.logger.LogCtx(ctx, "level", "debug", "message", "computed desired state")
-	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", "computed desired state", "app", appName)
 
-	return appCRs, nil
+	return appCR, nil
 }
 
-func (r *Resource) newAppCR(ctx context.Context, releaseCR releasev1alpha1.Release, component releasev1alpha1.ReleaseSpecComponent) (*applicationv1alpha1.App, error) {
+func (r *Resource) newAppCR(name, repository, version string) *applicationv1alpha1.App {
 	appCR := &applicationv1alpha1.App{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "App",
 			APIVersion: "application.giantswarm.io",
 		},
```

Comment:
We got a handy helper for this. See https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/application/v1alpha1/app_types.go#L53-L58.

Reply:
Nice